pkg/network/tracer/connection/ebpfless/payload.go

```diff
@@ -0,0 +1,116 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+// Package ebpfless contains supporting code for the ebpfless tracer
+package ebpfless
+
+import (
+	"fmt"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+
+	"github.com/DataDog/datadog-agent/pkg/network"
+)
+
+// Layers holds a set of network layers for a packet
+type Layers struct {
+	IP4 *layers.IPv4
+	IP6 *layers.IPv6
+	UDP *layers.UDP
+	TCP *layers.TCP
+}
+
+// NewLayers returns a new instance of `Layers`
+func NewLayers(family network.ConnectionFamily,
+	proto network.ConnectionType,
+	ip4 *layers.IPv4,
+	ip6 *layers.IPv6,
+	udp *layers.UDP,
+	tcp *layers.TCP,
+) Layers {
+	switch family {
+	case network.AFINET:
+		ip6 = nil
+	case network.AFINET6:
+		ip4 = nil
+	}
+
+	switch proto {
+	case network.TCP:
+		udp = nil
+	case network.UDP:
+		tcp = nil
+	}
+
+	return Layers{ip4, ip6, udp, tcp}
+}
```

Comment:
what will happen if `family` or `proto` are not supported?

Reply:
Added errors